Announce the funding team, Maintainer in Residence and RFMF launch (#1860)
rust-lang/blog.rust-lang.org: content/launching-the-rust-foundation-maintainer-fund.md

team_url = "https://www.rust-lang.org/governance/teams/leadership-council"
+++

> If you want to financially support the development of Rust, please consider [donating][rfmf-sponsors] to the Rust Foundation Maintainer Fund.

A few months ago, the Rust Foundation announced the [Rust Foundation Maintainer Fund][rfmf-announcement] (RFMF). Since then, the Rust Project has been closely cooperating with the Rust Foundation to determine how exactly this fund will be used to support Rust maintainers. This resulted in the acceptance of [RFC #3931][rfmf-rfc], which established the [Funding team][funding-team] and the [Maintainer in Residence][mir] program.

The primary goal of the Funding team is to ensure that maintainers who work on Rust and its toolchain will be properly supported. We will talk to Rust Project members to figure out their funding situation, meet Rust team leads to learn about their maintenance needs, approach companies to find opportunities for them to invest into Rust by supporting Rust maintainers, coordinate various funding efforts and ensure that the beneficial effects of funded maintenance are visibly promoted, with the help of the [Content team][content-team].

> [Maintainer in Residence][mir] is a new program dedicated to financially supporting existing Rust Project maintainers[^dir]. Each Maintainer in Residence will be funded to [maintain][maintenance-post] one or more critical parts of Rust, such as the compiler, the standard library, Cargo, Clippy or one of many other projects that the Rust Project develops and maintains. The funded work will include activities such as performing large-scale refactorings, code reviews, unblocking new features, issue triaging, mentoring other contributors and more, and will be split between priorities guided by the teams they are supporting and priorities of their own choosing within the Project.

Reviewer: We talked last week about having Rust Project Goals also being a place for funds to be directed towards. In the RFMF RFC, rereading it now, it does seem that whether that be directed directly or as a part of Maintainer in Residence there's some ties to the Rust Project Goals process we could add a short section or even a sentence on here and link out to the Rust Project Goals page.

What do y'all think?

```suggestion
[Maintainer in Residence][mir] is a new program dedicated to financially supporting existing Rust Project maintainers[^dir]. Each Maintainer in Residence will be funded to [maintain][maintenance-post] one or more critical parts of Rust, such as the compiler, the standard library, Cargo, Clippy or one of many other projects that the Rust Project develops and maintains. The funded work will include activities such as performing large-scale refactorings, code reviews, unblocking new features, issue triaging, mentoring other contributors and more, and will be split between priorities guided by the teams they are supporting and priorities of their own choosing within the Project. When the framing fits for the work, it's intended for the Maintainers in Residence to propose, pick up, and drive forward the completion of [Rust Project Goals](https://rust-lang.github.io/rust-project-goals/).
```

Reviewer 2: I said elsewhere we also have https://rust-lang.github.io/rust-project-goals/2026/help-wanted.html#looking-for-funds (though it's in the process of being revamped in https://github.com/rust-lang/rust-project-goals/pull/617, and t-funding should probably be involved in that)

Author: I went with this: "Where applicable, Maintainers in Residence are also encouraged to propose, champion, and drive forward [Rust Project Goals](https://rust-lang.github.io/rust-project-goals/)". Let me know what you think!

Author: > I said elsewhere we also have https://rust-lang.github.io/rust-project-goals/2026/help-wanted.html#looking-for-funds (though it's in the process of being revamped in [rust-lang/rust-project-goals#617](https://github.com/rust-lang/rust-project-goals/pull/617), and t-funding should probably be involved in that)

I'm up for linking to this initiative! As you said, I wasn't sure if it's already "active" or still WIP though.

Reviewer: I like how you wrote this @Kobzol 👍 

I'm good to resolve this, unless we'd like to keep it open until we link in what @lqd suggests.

Reviewer 2: We've had some form of "Help wanted", needing a contributor or funding, since the beginning, but not necessarily a dedicated page like this. Either way, the team should be involved in shaping that.

Author: Let me know what you think: https://github.com/rust-lang/blog.rust-lang.org/pull/1860/commits/e53057d72b328725b536c8ffea8bb272bb3de4cb

Reviewer 2: Looks good to me as well :) We can update this link if the page changes!

Reviewer: Let's mark this as resolved? 🤔

[^dir]: This program was inspired by the [Developer in Residence][developer-in-residence] concept used by the Python Software Foundation (PSF), with which we led several helpful discussions. Thank you, PSF!

This program extends our existing support for Rust maintainers, such as the [program management program](https://blog.rust-lang.org/inside-rust/2026/04/09/program-management-update-2026-03/) and the [compiler-ops program](https://blog.rust-lang.org/inside-rust/2025/06/05/a-glance-at-the-team-compiler-operations). An important development is that we now have a centralized [mechanism][rfmf-sponsors] for gathering donations from both individuals and companies, and a dedicated team that will help direct those funds to specific maintainers. You can find more details about the funding team and the Maintainer in Residence program in the [RFC][rfmf-rfc].

> We expect to hire the first Maintainer in Residence in the upcoming months and announce it on this blog post, so stay tuned!

Reviewer: Seemed like this would be a new blog post so "this" => "a"?

```suggestion
We expect to hire the first Maintainer in Residence in the upcoming months and announce it on a blog post, so stay tuned!
```

Reviewer 2: Or on _this_ blog?

Author: Ah, that should have been on this blog, thanks.

## How to contribute funds

> If you are an individual who wants to help Rust succeed and thrive, you can donate to RFMF through [GitHub Sponsors][rfmf-sponsors][^sponsors]. Companies who would like to invest in better maintenance of Rust can also donate through GitHub Sponsors or they can contact the Rust Foundation [directly](mailto:contact@rustfoundation.org).

Reviewer: ```suggestion
If you are an individual who wants to help Rust succeed and thrive, you can donate to the RFMF through [GitHub Sponsors][rfmf-sponsors][^sponsors]. Companies who would like to invest in better maintenance of Rust can also donate through GitHub Sponsors or they can contact the Rust Foundation [directly](mailto:contact@rustfoundation.org).
```

[^sponsors]: Note that the fact that GitHub Sponsors is currently enabled on the `rustfoundation` GitHub organization, and not the `rust-lang` organization, is an implementation detail that might change in the future. All donations raised on this Sponsors page will be routed to the Rust Foundation Maintainer Fund and will be spent on directly supporting Rust Project maintainers.

The important thing is that **all proceeds from this fund will be directly used to support Rust Project maintainers**. We currently expect that to happen primarily through the Maintainer in Residence program, but it can also be done in the form of smaller-scale grants or other mechanisms, as determined by the Funding team. We will figure this out on the go, as this is also quite new for us.

> We really appreciate each donation, however small, because with more money we can hire more maintainers to ensure that we can continue to develop Rust and that important improvements are not blocked on maintenance tasks. This is especially important at this time, where Rust is starting to get used more and more in the industry in various application areas, which increases the need for sustained maintenance, but at the same time we see that key Rust maintainers are losing their funding for Rust work due to budget shifts. The Rust Foundation Maintenance Fund is designed to provide stable funding for Rust maintainers that is less dependent on sudden shifts in the job market and the IT industry.

Reviewer: Trying to tease apart the ideas here; sentence felt a bit long.

```suggestion
We really appreciate each donation, however small, because with more money we can hire more maintainers to ensure that we can continue to develop Rust and that important improvements are not blocked on maintenance tasks. This is especially important at this time, where Rust is starting to get used more and more in the industry in various application areas, which increases the need for sustained maintenance. The importance of all funding sources is underscored by a trend we see, where key Rust maintainers are losing their funding for Rust work due to budget shifts. The Rust Foundation Maintenance Fund is designed to provide stable funding for Rust maintainers that is less dependent on sudden shifts in the job market and the IT industry.
```

> As with most things, there is no one-size-fits-all solution for funding maintainers, and there are multiple ways to support Rust. The [RustNL Maintainers Team][rustnl-maintainers] recently hired several Rust Project contributors, and we previously [wrote][individual-contributors] about how you can support specific individuals working on Rust. We welcome all efforts that can help support Rust Project maintainers, who often do work that is near invisible and thankless, although at the same time incredibly important and necessary, on a volunteer basis.

Reviewer: ```suggestion
As with most things, there is no one-size-fits-all solution for funding maintainers, and there are multiple ways to support Rust. The [RustNL Maintainers Team][rustnl-maintainers] recently hired several Rust Project contributors, and we previously [wrote][individual-contributors] about how you can support specific individuals working on Rust. We welcome all efforts that can help support Rust Project maintainers, who often do work that is near invisible and thankless, while at the same time incredibly important and necessary, on a volunteer basis.
```
